packages/core/src/components/drawer/drawer.tsx

```diff
@@ -0,0 +1,161 @@
+import { IBackdropProps, IOverlayableProps, Overlay } from "../overlay/overlay";
+
+export interface IDrawerProps extends IOverlayableProps, IBackdropProps, IProps {
+    /**
+     * Toggles the visibility of the overlay and its children.
+     * This prop is required because the component is controlled.
+     */
+    isOpen: boolean;
+
+    /**
+     * Drawer always has a backdrop so this prop is excluded from the public API.
+     * @internal
+     */
+    hasBackdrop?: boolean;
+
+    /**
+     * Name of a Blueprint UI icon (or an icon element) to render in the
+     * dialog's header. Note that the header will only be rendered if `title` is
+     * provided.
+     */
+    icon?: IconName | JSX.Element;
+
+    /**
+     * Whether to show the close button in the dialog's header.
+     * Note that the header will only be rendered if `title` is provided.
+     * @default true
+     */
+    isCloseButtonShown?: boolean;
+
+    /**
+     * CSS styles to apply to the dialog.
+     * @default {}
+     */
+    style?: React.CSSProperties;
+
+    /**
+     * Title of the dialog. If provided, an element with `Classes.DIALOG_HEADER`
+     * will be rendered inside the dialog before any children elements.
+     */
+    title?: React.ReactNode;
+
+    /**
+     * Name of the transition for internal `CSSTransition`. Providing your own
+     * name here will require defining new CSS transition properties.
+     */
+    transitionName?: string;
+
+    /**
+     * Whether the drawer should appear with vertical styling.
+     * @default false
+     */
+    vertical?: boolean;
+
+    /** Whether this drawer should use large styles. */
+    large?: boolean;
+
+    /** Whether this drawer should use small styles. */
+    small?: boolean;
+}
+
+export class Drawer extends AbstractPureComponent<IDrawerProps, {}> {
+    public static defaultProps: IDrawerProps = {
+        canOutsideClickClose: true,
+        isOpen: false,
+        large: false,
+        small: false,
+        vertical: false,
+    };
+
+    public static displayName = `${DISPLAYNAME_PREFIX}.Drawer`;
+
+    public render() {
+        const classes = classNames(
+            Classes.DRAWER,
+            {
+                [Classes.LARGE]: this.props.large,
+                [Classes.SMALL]: this.props.small,
+                [Classes.VERTICAL]: this.props.vertical,
+            },
+            this.props.className,
+        );
+        // small drawer should not use a backdrop
+        if (this.props.small !== false) {
+            return (
+                <Overlay {...this.props} className={Classes.OVERLAY_CONTAINER} hasBackdrop={false}>
+                    <div className={classes} style={this.props.style}>
+                        {this.maybeRenderHeader()}
+                        {this.props.children}
+                    </div>
+                </Overlay>
+            );
+        }
+        return (
+            <Overlay {...this.props} className={Classes.OVERLAY_CONTAINER} hasBackdrop={true}>
```

Comment:
allow user to override, with default behavior. 
```suggestion
            <Overlay {...this.props} className={Classes.OVERLAY_CONTAINER} hasBackdrop={this.props.hasBackdrop || !this.props.small}>
```
then remove `if` block above.